main.go

```diff
@@ -84,6 +85,22 @@ func init() {
 	flag.StringVar(&config.YoukuPassword, "password", "", "Youku password")
 	// youtube
 	flag.BoolVar(&config.YouTubeStream2, "ytb-stream2", false, "Use data in url_encoded_fmt_stream_map")
+	var singleThreadSpeedLimit string
+	flag.StringVar(&singleThreadSpeedLimit, "singleThreadSpeedLimit", "", "Limit the speed of each thread, 0 is unlimited")
+
+	flag.Parse()
+
+	if singleThreadSpeedLimit == "" || singleThreadSpeedLimit == "0" {
```

Comment:
Move flag processing to the main method? Or do you have a special reason that it has to be there?

Reply:
There is no special meaning. Previously, config was initialized in init function, so I move it here. Is it better to put this logic inside the main function?

Reply:
Well you removed the flag.parse from main and moved it out. But before it was in main, because at the start of main logic depending on flags is executed. So for me it would make more sense to add the logic back into main. Or am I missing something here?

Reply:
If don't execute `flag.Parse()`, singleThreadSpeedLimit will always be `""`, so I moved it here.

Reply:
https://github.com/muzea/annie/pull/1

Reply:
> There is no special meaning. Previously, config was initialized in init function, so I move it here. Is it better to put this logic inside the main function?


Well it was in main method, because after the flag.parse() are the check that are depending on the value in the flag fields.

So you can put the flag.parse() back into the main method and move your coding below that statement like, I recommended in the commit posted above. With that you don't change the structure of the coding :)